sdk/eventhubs/azure-messaging-eventhubs/TROUBLESHOOTING.md

```diff
@@ -0,0 +1,260 @@
+  - [Enable AMQP transport logging](#enable-amqp-transport-logging)
+  - [Reduce logging](#reduce-logging)
+- [Troubleshoot EventProducerAsyncClient/EventProducerClient issues](#troubleshoot-eventproducerasyncclienteventproducerclient-issues)
+  - [Cannot set multiple partition keys for events in EventDataBatch](#cannot-set-multiple-partition-keys-for-events-in-eventdatabatch)
+  - [Setting partition key on EventData is not set in Kafka consumer](#setting-partition-key-on-eventdata-is-not-set-in-kafka-consumer)
+- [Troubleshoot EventProcessorClient issues](#troubleshoot-eventprocessorclient-issues)
+  - [412 precondition failures when checkpointing](#412-precondition-failures-when-checkpointing)
+  - [Partition ownership changes a lot](#partition-ownership-changes-a-lot)
+  - ["...current receiver 'nil' with epoch '0' is getting disconnected"](#current-receiver-nil-with-epoch-0-is-getting-disconnected)
+  - [High CPU usage](#high-cpu-usage)
+  - [Processor client stops receiving](#processor-client-stops-receiving)
+  - [Migrate from legacy to new client library](#migrate-from-legacy-to-new-client-library)
+- [Get additional help](#get-additional-help)
+
+## Handle Event Hubs exceptions
+
+All Event Hubs exceptions are wrapped in an [AmqpException][AmqpException].  They often have an underlying AMQP error code which specifies whether an error is retryable or not.  For retryable errors (ie. "amqp\:connection\:forced" or "amqp\:link\:detach-forced"), the client libraries will attempt to recover from these errors based on the [retry options][AmqpRetryOptions] specified when instantiating the client.  To configure retry options, follow the sample [Publish events to specific partition][PublishEventsToSpecificPartition].  If the error is non-retryable, there is some configuration issue that the customer needs to resolve.
+
+The recommended way to solve the specific exception the AMQP exception represents is to follow the
+[Event Hubs Messaging Exceptions][EventHubsMessagingExceptions] guidance.
+
+### Find relevant information in exception messages
+
+An [AmqpException][AmqpException] contains three fields which describe the error.
+
+* **getErrorCondition**: The underlying AMQP error. A description of the errors can be found in the [AmqpErrorCondition][AmqpErrorCondition] javadocs or the OASIS AMQP 1.0 spec.
+* **isTransient**: Whether or not trying to perform the same operation is possible.  SDK clients apply the retry policy when the error is transient.
+* **getErrorContext**: Information about where the AMQP error originated.
+  * [LinkErrorContext][LinkErrorContext]: Errors that occur in either the send/receive link.
+  * [SessionErrorContext][SessionErrorContext]: Errors that occur in the session.
+  * [AmqpErrorContext][AmqpErrorContext]: Errors that occur in the connection or a general AMQP error.
+
+### Commonly encountered exceptions
+
+#### amqp\:connection\:forced and amqp\:link\:detach-forced
```

Comment:
The service will also boot connections every 30 minutes, whether idle or not... not sure if that's significant in this context, as if the client is active, nobody should notice.   However, it may appear in the logs?

Reply:
I had no idea it did that... hmm.